libkbfs/identify_util.go

```diff
@@ -170,6 +177,34 @@ func identifyUID(ctx context.Context, nug normalizedUsernameGetter, identifier i
 	return nil
 }
 
+// identifyUserToChan calls identifyUser and plugs the result into the error channnel.
+func identifyUserToChan(ctx context.Context, nug normalizedUsernameGetter,
+	identifier identifier, name libkb.NormalizedUsername, uid keybase1.UID,
+	isPublic bool, errChan chan error) {
+	errChan <- identifyUser(ctx, nug, identifier, name, uid, isPublic)
+}
+
+// identifyUsers identifies the users in the given maps.
+func identifyUsers(ctx context.Context, nug normalizedUsernameGetter,
+	identifier identifier, writers, readers map[keybase1.UID]libkb.NormalizedUsername,
+	public bool) error {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	nitems := len(writers) + len(readers)
+	errChan := make(chan error, nitems)
+	// TODO: limit the number of concurrent identifies? Otherwise we can use
+	// errgroup.Group here.
+	for uid, name := range writers {
+		go identifyUserToChan(ctx, nug, identifier, name, uid, public, errChan)
```

Comment:
Hmm, might be better to just use errgroup.Group here, since it has the advantage that it cancels in-flight goroutines if an error is encountered.

You can leave the TODO in, and add a comment about writing a concurrency-limited version of errgroup.Group.